### PR TITLE
Stratification

### PR DESCRIPTION
**Pull request by bikramkawan**
The stratification based on new layout.

### CODE REVIEW

**Comment by mstreit** [comment]
When I add two categorical attributes, switch the stratification to the second, and then add a numerical attribute, the stratification also switches back to the first categorical attribute. @bikramkawan, please fix.

**Comment by mstreit** [comment]
The selected stratification is also set back to the initial one when I resort any column.

**Comment by mstreit** [comment]
The stratification reset problem is fixed. The re-sort problem is still there.

**Comment by thinkh** [comment]
@mstreit I can't see any problem with the re-sort. Could you please write the steps to reproduce?

**Comment by mstreit** [comment]
it works. you can merge.

File: src/filter/CategoricalFilter.ts [hunk 1]
```diff
@@ -39,6 +42,23 @@ export default class CategoricalFilter extends AVectorFilter<string, ICategorica
...
   }
 
+  private attachListener() {
+    const splitIcon = this.$node.select('header').insert('a', ':first-child')
+      .classed('fa fa-bars', true);
+    splitIcon.on('click', () => {
+      d3.selectAll('.fa.fa-bars').style('border', null);
+      const b = splitIcon.attr('class');
+      if (b === 'fa fa-bars') {
+        fire(CategoricalColumn.EVENT_STRATIFYME, this.data.desc.id);
```
Comment (thinkh) [inline src/filter/CategoricalFilter.ts]: Fire `column` instead of `colId`, as we do for sort event.

File: src/filter/CategoricalFilter.ts [hunk 2]
```diff
@@ -39,6 +42,23 @@ export default class CategoricalFilter extends AVectorFilter<string, ICategorica
...
+        splitIcon.style('border', '1px solid');
```
Comment (thinkh) [inline src/filter/CategoricalFilter.ts]: Create CSS class and toggle with `.classed('myClass', true)`

File: src/column/ColumnManager.ts [hunk 3]
```diff
@@ -39,20 +40,24 @@ export default class ColumnManager extends EventHandler {
...
   static readonly EVENT_DATA_REMOVED = 'removedData';
 
-  private $node:d3.Selection<any>;
+  private $node: d3.Selection<any>;
 
   readonly columns: AnyColumn[] = [];
   private filtersHierarchy: AnyColumn[] = [];
   private firstColumnRange: Range;
-  private rangeList = [];
+  private stratifiedRanges: Range[];
```
Comment (thinkh) [inline src/column/ColumnManager.ts]: Please add a short description

**Pull request merged**